source/slang/slang-emit-c-like.cpp

```diff
@@ -674,6 +674,18 @@ String CLikeSourceEmitter::generateName(IRInst* inst)
         return String(intrinsicDecoration->getDefinition());
     }
 
+    // If the instruction reprsents one of the "magic" declarations
```

Comment:
We probably do want a modifier at some point that indicates that a declaration should get a mangled name using rules similar to `extern "C"`, and optionally even a way to specify exactly what the mangled name should be.

The catch is that such a feature wouldn't really help in this case, because our output HLSL/GLSL (and almost all our output code, really) doesn't use what the Slang compiler thinks of as mangled names at all.

Instead, most of the names you see in output code are the result of taking the "name hint" decorations on instructions and then making them unique (where "making unique" is not the same as mangling). An earlier version of this change (while I was just getting things roughed out) used the name hint decoration and simply skipped the unique-ing step, but that would run into the problem that we lose all name hints in obfuscated code.

(Aside: it is worth nothing that this change means that the names of NVAPI-relevant constructs will appear as clear text even in obfuscated output, but that seems reasonable since we also do the same for things like builtin/intrinsic functions that are called from obfuscated code. The fact that code uses NVAPI is not something we can easily hide, given the constraints.)